indico/modules/events/sessions/controllers/management/sessions.py

```diff
@@ -123,3 +129,49 @@ def _process_PATCH(self):
             updates['colors'] = colors
         update_session(self.session, updates)
         return jsonify()
+
+
+class RHSessionPersonList(RHManageSessionsActionsBase):
+    """List of persons in the session"""
+
+    def _process(self):
+        session_ids = {s.id for s in self.sessions}
+        session_persons = (ContributionPersonLink
+                           .find(ContributionPersonLink.contribution.has(Contribution.session_id.in_(session_ids)))
+                           .all())
+        session_persons.extend(SubContributionPersonLink
+                               .find(SubContributionPersonLink.subcontribution
+                                     .has(SubContribution.contribution.has(Contribution.session_id.in_(session_ids))))
+                               .all())
+        return jsonify_template('events/sessions/management/session_person_list.html',
```

Comment:
The query gives you duplicates since there might be multiple links to the same person. `.distinct(ContributionPersonLink.person_id)` would get rid of them, but then you'd lose information in case of someone who's a speaker in contribution 1 but an author in contribution 2.

You can simply iterate over `session_persons` and build a dict (actually this is the perfect case for a `defaultdict`) mapping persons to the other information (speaker/primary-author/secondary-author). That way you don't end up with duplicates but still show all the states (and in the future it could be easily extended to list the linked contributions to show them e.g. in a tooltip)